File: 3기/디자인패턴/최혁/workspace/Visitor/src/File.java

public class File extends Entry {

	private String name;
	private int size;

	public File(String name, int size) {
		this.name = name;
		this.size = size;
	}

	@Override
	public String getName() {
		return name;
	}

	@Override
	public int getSize() {
		return size;
	}

	@Override
	public void accept(Visitor v) {
		v.visit(this);
	}

//	@Override
//	protected void printList(String prefix) {
//		System.out.println(prefix + "/" + this); // this == this.toString()
//	}
	

}

File: 3기/디자인패턴/최현주/SourceCode/CompositePattern/File.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace CompositePattern.ex2
{
    /// <summary>
    /// Leaf 역할 (단일 객체)
    /// Entry(Component)를 구현하는 클래스.
    /// </summary>
    public class File : Entry
    {
        private string name;
        private int size;

        public File(string name, int size)
        {
            this.name = name;
            this.size = size;
        }

        public override string GetName()
        {
            return name;
        }

        public override int GetSize()
        {
            return size;
        }

        public override void PrintList(string prefix)
        {
            Debug.Log(prefix + "/" + this.name);
        }
    }
}